### ISSUE TITLE
Discussion: Remove possibility to change classifier

### ISSUE THREAD

**Issue by k-dominik** [ilastik/ilastik#2336]
Pixel Classification/Autocontext ... allow to change the classifier. We currently do not test this functionality and we've  heard of problems [on the forum](ilastik ilastik/ilastik@a7ee78d7936ed573b10c6ff4d8733e1f126a00ea
volumina ilastik/volumina@64b12218163e6fa9b7ccde6460ee2f79c93f2bdd
), #2335 

[Benefit of this functionality](https://jmlr.org/papers/v15/delgado14a.html) is questionable, and produces more work in maintenance.  IMO we should _at least_ move this to debug mode, if not remove it completely. Same goes for the feature importance table, which is also of questionable value.

**Comment by Tomaz-Vieira** [comment]
Would this break old .ilp files that were created with any of the deprecated classifiers?

**Comment by k-dominik** [comment]
Removing only the _possibility_ to change classifiers (so basically the menu entry) would preserve compatibility with old project files. My main issue with this "feature"  is that there is an unclear advantage to not having it at all. I think it's there only because we can, not that anyone really (should) need it.

**Comment by m-novikov** [comment]
I agree as this functionality is not tested it should not be available for an end user.
For debug mode it maybe ok.
This functionality is only useful for exploratory purposes.

**Comment by m-novikov** [comment]
We could hide it behind a feature flag(configuration) in the new releases to see how many people actually use it.
